### PR TITLE
CONSOLE-3351: Propagate hub-cluster's URL to managed clusters

### PR DESCRIPTION
**Pull request by jhadvig**
Story: https://issues.redhat.com/browse/CONSOLE-3351

This PR addressed two mechanisms:

1.  Creating `hub-cluster-config` CM on all the manages cluster in the fleet using the MCAction. This CM will contain a single entry `consoleBaseAddress: <hub-cluster's-console-url>`.
2. Read this CM by all the managed clusters in the fleet and set the hub-cluster's URL on the console-config CM that is used for configuration of the console server

/assign @TheRealJon

### CODE REVIEW

File: pkg/console/subresource/consoleserver/types.go [hunk 1]
```diff
@@ -29,9 +29,14 @@ type Config struct {
 	I18nNamespaces           []string          `yaml:"i18nNamespaces,omitempty"`
 	Proxy                    Proxy             `yaml:"proxy,omitempty"`
 	ManagedClusterConfigFile string            `yaml:"managedClusterConfigFile,omitempty"`
+	HubClusterInfo           `yaml:"hubClusterInfo,omitempty"`
 	Telemetry                map[string]string `yaml:"telemetry,omitempty"`
 }
 
+type HubClusterInfo struct {
```
Comment (jhadvig) [inline pkg/console/subresource/consoleserver/types.go:36]: not really sure about this struct, if we wanna have a higher abstraction here, eg. `MultiClusterInfo`, which would contain all configuration that we might add down the road as well as `HubClusterInfo`.

Thoughts ?

File: pkg/api/api.go [hunk 2]
```diff
@@ -10,54 +10,58 @@ const (
...
+	ManagedClusterAPIServerCertMountDir            = "/var/managed-cluster-api-server-certs"
+	ManagedClusterAPIServerCertName                = "managed-cluster-api-server-cert"
+	ManagedClusterAPIServerCertKey                 = "ca-bundle.crt"
+	ManagedClusterOAuthServerCertMountDir          = "/var/managed-cluster-oauth-server-certs"
+	ManagedClusterOAuthServerCertName              = "managed-cluster-oauth-server-cert"
+	ManagedClusterOAuthServerCertKey               = "ca-bundle.crt"
+	ManagedClusterOAuthClientName                  = "console-managed-cluster-oauth-client"
+	OAuthClientManagedClusterViewName              = "console-oauth-client"
+	CreateOAuthClientManagedClusterActionName      = "console-create-oauth-client"
+	CreateHubClusterConfigManagedClusterActionName = "console-hub-cluster-config"
```
Comment (jhadvig) [inline pkg/api/api.go:62]: This one was added 👆

File: pkg/api/api.go [hunk 3]
```diff
@@ -10,54 +10,58 @@ const (
...
+	OpenShiftCustomLogoConfigMapName               = "custom-logo"
+	TrustedCAConfigMapName                         = "trusted-ca-bundle"
+	TrustedCABundleKey                             = "ca-bundle.crt"
+	TrustedCABundleMountDir                        = "/etc/pki/ca-trust/extracted/pem"
+	TrustedCABundleMountFile                       = "tls-ca-bundle.pem"
+	OCCLIDownloadsCustomResourceName               = "oc-cli-downloads"
+	ODOCLIDownloadsCustomResourceName              = "odo-cli-downloads"
+	HubClusterName                                 = "local-cluster"
+	HubClusterConfigMapName                        = "hub-cluster-config"
+	HubClusterConsoleURLKey                        = "consoleBasePath"
```
Comment (jhadvig) [inline pkg/api/api.go:48]: These two were added 👆

File: pkg/api/api.go [hunk 4]
```diff
@@ -10,54 +10,58 @@ const (
...
-	CreateOAuthClientManagedClusterActionName = "console-create-oauth-client"
-	OAuthServerCertManagedClusterViewName     = "console-oauth-server-cert"
-	OLMConfigManagedClusterViewName           = "olm-config"
+	ClusterOperatorName                            = "console"
+	OpenShiftConsoleName                           = "console"
+	OpenShiftConsoleNamespace                      = TargetNamespace
+	OpenShiftConsoleOperatorNamespace              = "openshift-console-operator"
+	OpenShiftConsoleOperator                       = "console-operator"
+	OpenShiftConsoleConfigMapName                  = "console-config"
+	OpenShiftConsoleConfigMapKey                   = "console-config.yaml"
```
Comment (jhadvig) [inline pkg/api/api.go:19]: This one was added 👆

File: pkg/console/subresource/configmap/managed_clusters.go [hunk 5]
```diff
@@ -34,3 +36,23 @@ func ManagedClustersConfigMapStub() *corev1.ConfigMap {
...
+
+	return configMap, nil
+}
+
+func HubClusterConfigMapStub() *corev1.ConfigMap {
+	configMap := ConsoleConfigMapStub()
+	configMap.Name = api.HubClusterConfigMapName
+	configMap.Labels = util.SharedLabels()
+	configMap.Labels[api.ManagedClusterLabel] = ""
+	return configMap
```
Comment (TheRealJon) [inline pkg/console/subresource/configmap/managed_clusters.go:55]: bindata?

**Review by TheRealJon — COMMENTED** [review]
We should probably go ahead and use the ManifestWork API since I'm already working on removing the ManagedClusterAction API in another story.

**Comment by spadgett** [comment]
> We should probably go ahead and use the ManifestWork API since I'm already working on removing the ManagedClusterAction API in another story.

+1 let's use ManifestWork for this change

**Comment by jhadvig** [comment]
Closing based on the discussion that this is not the UX approach we want

**Pull request closed without merging**